### PR TITLE
Add stockfish support

### PR DESCRIPTION
**Pull request by ThanosApostolou**
I have only test the x86_64 build. Unfortunately stockfish has only a weird makefile which needs the `ARCH` variable to be specified accordingly to the architecture. I don't see in the makefile any support for aarch64 but maybe the armv7 build can work as well, or maybe we can override the other variables of the makefile to get a aarch64 build. Is there any way I can test it in aarch64 (I haven't found any flatpak guide for compiling and testing in different architecture apart i386).

### CODE REVIEW

File: org.gnome.Chess.json [hunk 1]
```diff
@@ -13,6 +13,45 @@
...
+                        }
+                    },
+                    "arm": {
+                        "env": {
+                            "ARCH": "armv7"
+                        }
+                    },
+                    "aarch64": {
+                        "env": {
+                            "ARCH": "armv7"
```
Comment (mcatanzaro) [inline org.gnome.Chess.json]: Looks like this worked; at least the aarch64 build succeeded. It's pretty evil though; would be better to patch the Makefile.

I wonder why Stockfish is using a custom Makefile instead of a proper build system. :/

**Comment by nedrichards** [comment]
If you're concerned about testing you can make the module only target a specific arch.

**Comment by ThanosApostolou** [comment]
@nedrichards I thought that this was possible only for building i386 from x86_64 pc but I found this:
https://mathieu.daitauha.fr/blog/2017/08/09/building-apps-for-other-arches-with-flatpak/
Unfortunately I tried to test the aarch64 myself but I stopped because it takes way too much time to build on my machine (probably because of qemu emulation).

@Zlopez patching the makefile is more difficult and messier because it's using the `ARCH` variable in order to set other variables and cxxflags with many if statements which is hard to follow. So, I suggest that we use it like this for now and I'll report an issue upstream in order to ask some nicer method to build stockfish (or at least to provide an aarch64 option too).

**Comment by ThanosApostolou** [comment]
I took the opportunity to add phalanx too since someone asked it and it was really simple. Tested them and they work good. Probably pbook.phalanx and sbook.phalanx are not need to be installed, but the official makefile has them at install target so I thought that maybe they are needed for something.

**Comment by Zlopez** [comment]
I tested the phalanx and it's working. I will merge it now.

@ThanosApostolou FYI @mcatanzaro is maintainer of upstream GNOME Chess

**Comment by ThanosApostolou** [comment]
Aaaa ok I didn't know :stuck_out_tongue: Nice to see upstream developers taking close look at flathub packages (hopefully in time this will expand to other developers too apart of gnome ones).

Anyway thx for accepting this, I created an issue at stockfish github about adding a build system or at least an aarch64 option at the makefile. So, I will contact you if they do.

**Comment by mcatanzaro** [comment]
> I took the opportunity to add phalanx too since someone asked it and it was really simple. Tested them and they work good. Probably pbook.phalanx and sbook.phalanx are not need to be installed, but the official makefile has them at install target so I thought that maybe they are needed for something.

Those are opening books... at the beginning of games, the engine looks up where to move from the books (database) instead of doing its normal AI thing (computing future possibilities).

From https://sources.debian.org/src/phalanx/22+d051004-14/phalanx.h/#L182 it looks like there are a bunch of semi-hardcoded paths here, not exactly great code. So the book usage is probably broken. Looks like we can probably do `make PBOOKDIR=/app/lib SBOOKDIR=/app/lib` in build-commands. But then the LEARN_DIR is more complicated. That is a learning book, which is writable. So it can't go under /app/lib, it really needs to go in the user's home directory. But of course that's not possible to do by defining LEARN_DIR at build time. openSUSE has a patch to fix this at https://build.opensuse.org/package/show/openSUSE:Factory/phalanx, but, terrifyingly, it also has a patch I added to fix castling from 2013 still applied, even though the current version XXV is several years after that should have gone upstream.

So my conclusion is: maybe it would be easier to just not include Phalanx, oops. Sorry. Anyway, up to you. I would test castling at least, to make sure you don't hit https://bugzilla.opensuse.org/show_bug.cgi?id=819525.

**Comment by mcatanzaro** [comment]
> So my conclusion is: maybe it would be easier to just not include Phalanx, oops. Sorry. Anyway, up to you. I would test castling at least, to make sure you don't hit https://bugzilla.opensuse.org/show_bug.cgi?id=819525.

Also note the reproducer I have there for the castling bug *probably* won't work unless the engine is using its opening books, though there's only one way to find out for sure. ;)

**Comment by ThanosApostolou** [comment]
Hmm I see.., I did have some cases that phalanx stopped working, but not after castling... and I had 3 successful complete games with castling using phalanx. But yeah I see that the books are not used when running
```
flatpak run --command=bash org.gnome.Chess
```
and then
```
[📦 org.gnome.Chess ~]$ phalanx
Phalanx XXV
telluser Phalanx: Cannot open both [sbook.phalanx] and [pbook.phalanx]
tellics set 1 Phalanx XXV, 820 kB hashtable, 0/0 kB P/S opening book
```

I'll play with it and see if I can fix it. I'll do a new pull request if I am successful.

**Comment by mcatanzaro** [comment]
> Hmm I see.., I did have some cases that phalanx stopped working,

Probably better not to package it then. Or, if you're willing to do some debugging, try to figure out what's going wrong. You can use `G_MESSAGES_DEBUG=all` to print the communication between gnome-chess and the chess engine, which should reveal the problem when Phalanx breaks.

**Comment by Zlopez** [comment]
I see that I merged this too soon, but if @ThanosApostolou will be able to fix it soon I will accept the fix. If not I will just remove phalanx.
